pkg/proxy/ipvs/netlink_linux.go

```diff
@@ -143,52 +148,63 @@ func (h *netlinkHandle) ListBindAddress(devName string) ([]string, error) {
 // 10.0.0.10  scope host  src 10.0.0.10
 // Then cut the unique src IP fields,
 // --> result set: [10.0.0.1, 10.0.0.10]
-
-// If filterDev is specified, the result will discard route of specified device and cut src from other routes.
 func (h *netlinkHandle) GetLocalAddresses(dev, filterDev string) (sets.String, error) {
-	chosenLinkIndex, filterLinkIndex := -1, -1
 	if dev != "" {
-		link, err := h.LinkByName(dev)
-		if err != nil {
-			return nil, fmt.Errorf("error get device %s, err: %v", dev, err)
-		}
-		chosenLinkIndex = link.Attrs().Index
-	} else if filterDev != "" {
-		link, err := h.LinkByName(filterDev)
-		if err != nil {
-			return nil, fmt.Errorf("error get filter device %s, err: %v", filterDev, err)
+		if filterDev != "" {
+			return nil, fmt.Errorf("Invalid argument")
 		}
-		filterLinkIndex = link.Attrs().Index
+		return h.getLocalAddressesDev(dev)
+	}
+
+	if filterDev == "" {
```

Comment:
It seems like empty filterDev was valid but now we're erroring here. Do you know if we never pass in empty filterDev here?

Reply:
AFAIk this is only used in https://github.com/kubernetes/kubernetes/blob/e6922078577c5c8896b3619857dd744f1c1ecf6b/pkg/proxy/ipvs/proxier.go#L305-L324

Reply:
I have now made the _really big_ refactoring that I wanted to do from the start. I do not like "combo" functions with clever use of parameters to make the function do different things. @andrewsykim IMHO your question proves my point.

@aojea Sorry, I have removed the big address comment again. It is no longer relevant.